pkg/apis/flowcontrol/validation/validation.go

```diff
@@ -273,28 +278,55 @@ func ValidatePriorityLevelConfiguration(pl *flowcontrol.PriorityLevelConfigurati
 func ValidatePriorityLevelConfigurationSpec(spec *flowcontrol.PriorityLevelConfigurationSpec, name string, fldPath *field.Path) field.ErrorList {
 	var allErrs field.ErrorList
 	switch spec.Type {
-	case flowcontrol.PriorityLevelQueuingTypeExempt:
-		if spec.Queuing != nil {
-			allErrs = append(allErrs, field.Forbidden(fldPath.Child("queuing"), "must be nil if the type is not Queuing"))
+	case flowcontrol.PriorityLevelEnablementExempt:
+		if spec.Limited != nil {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("limited"), "must be nil if the type is not Limited"))
 		}
-	case flowcontrol.PriorityLevelQueuingTypeQueueing:
-		if spec.Queuing == nil {
+	case flowcontrol.PriorityLevelEnablementLimited:
+		if spec.Limited == nil {
```

Comment:
we should consider eliminate this nil check by assigning an empty struct if the type is "Limited" __and__ the field is a nil pointer, so that the inner defaulting can work. otherwise this validation is forcing users to set an empty struct for the field even if it doesn't really make a difference from nil pointer:
```
spec:
   type: Limited
   limited: {} # we should not force users to set this empty struct 
               # if it can be fully defaulted
```

Reply:
I agree